### PR TITLE
Voter reg status endpoints

### PR DESCRIPTION
**Pull request by cap10morgan**
[Pivotal card](https://www.pivotaltracker.com/story/show/101545426)

Tests pass, docker-compose builds and runs.

Assigning to Troy because he's building the other end in voter-registration-works.

### CODE REVIEW

File: resources/config.edn [hunk 1]
```diff
@@ -5,4 +5,7 @@
                          :port #resource-config/edn #resource-config/env "RABBITMQ_PORT_5672_TCP_PORT"}
             :queues {"voter-registration-http-api.ok" {:exclusive false :durable true :auto-delete false}
                      "voter-registration-works.registration-methods.read" {:exclusive false :durable true :auto-delete false}
-                     "voter-registration-works.voter.register" {:exclusive false :durable true :auto-delete false}}}}
+                     "voter-registration-works.voter.register" {:exclusive false :durable true :auto-delete false}
+                     "voter-registration-works.registration-statuses.read" {:exclusive false :durable true :auto-delete false}
```
Comment (tank157) [inline resources/config.edn]: The pluralization of status here is a deviation from the norm. Can to weigh in on why? And why, for instance, we would do it here, but not make a similar pluralization of the http endpoint?

File: src/voter_registration_http_api/service.clj [hunk 2]
```diff
@@ -77,6 +77,81 @@
...
+  (interceptor
+   {:enter
+    (fn [ctx]
+      (let [user-id (java.util.UUID/fromString
+                     (get-in ctx [:request :path-params :user-id]))
+            source (get-in ctx [:request :path-params :source])
+            result-chan (vrw/registration-statuses-read
+                         (merge {:user-id user-id}
+                                (when source {:source (keyword source)})))]
+        (go
```
Comment (tank157) [inline src/voter_registration_http_api/service.clj]: Sure seems like a lot of almost identical code in the go blocks in these interceptors. Not saying we need to do anything about it now, but could be an opportunity to isolate what's different and streamline the common code into some kind of helper.

File: resources/config.edn (hunk 1 above)
Comment (cap10morgan) [inline resources/config.edn]: I agree that the queue name should be singular. This came from the endpoint that returns a vector of them, but that doesn't mean the queue should be named like this. I'll change it.

File: src/voter_registration_http_api/service.clj (hunk 2 above)
Comment (cap10morgan) [inline src/voter_registration_http_api/service.clj]: I have a Friday project I'll be PR'ing today or tomorrow (hopefully) that eliminates this boilerplate. I agree that it's high time to do so.

File: test/voter_registration_http_api/service_test.clj [hunk 3]
```diff
@@ -238,3 +238,116 @@
...
+          [response-ch message] (async/alt!! channels/registration-statuses-read ([v] v)
+                                             (async/timeout 1000) [nil ::timeout])
+          response {:status :ok
+                    :registration-statuses [{:user-id user-id
+                                             :status :foo
+                                             :source :bar}
+                                            {:user-id user-id
+                                             :status :baz
+                                             :source :qux}]}]
+      (assert (not= message ::timeout))
```
Comment (tank157) [inline test/voter_registration_http_api/service_test.clj]: Ah, nice, I need to be doing this more often, not just doing an `(is (not= :timeout ...)`

Comment (cap10morgan) [inline test/voter_registration_http_api/service_test.clj]: @ericnormand suggested this while I was pairing with him. I think the reasoning was something like: "It makes no sense to run the rest of the checks if this fails, so let's blow up here."

Comment (tank157) [inline test/voter_registration_http_api/service_test.clj]: Yeah, that's exactly why I like it, too often I've had failing tests that just should have stopped at the timeout.

**Comment by tank157** [comment]
LGTM, test pass, runs inside it's own docker-compose without failing, didn't try it in the stein though, but that can wait until other parts are ready.

**Pull request merged**